protostar/commands/test/test_command.py

```diff
@@ -134,6 +151,14 @@ async def run(self, args) -> TestingSummary:
             seed=args.seed,
             slowest_tests_to_report_count=args.report_slowest_tests,
         )
+        failed_tests_paths = [
+            str(failed_test.file_path.absolute()) for failed_test in summary.failed
+        ]
+        protostar_cache.persist(
+            self._project_root_path,
+            "test_results",
+            {"failed_tests": failed_tests_paths},
```

Comment:
It looks like all test cases in test suite will be re-run.

Reply:
[rerun also broken tests](https://github.com/software-mansion/protostar/pull/895/commits/0fa195f8e910178b5ea50f990ab480141d06497a)

Reply:
I don't understand, sorry. I'm picking only the failed/broken tests. The results are overridden every time.

Reply:
`failed_test.file_path` returns a path to the test suite (e.g. `tests/test_main.cairo`). A test suite can have multiple test cases. If one test cases fails, `protostar test -lf` will re-run all tests in that test suite.

Reply:
Ah, right, I get it now, thx!

Reply:
Done in [recognize failed cases, not the entire suites](https://github.com/software-mansion/protostar/pull/895/commits/72db681232b493fee21d2a0097700e43030e2349)